### PR TITLE
fix: fix resource version api

### PR DESCRIPTION
**Pull request by Han-Ya-Jun**
### Description
查询资源版本数据兼容历史数据以及返回资源插件信息

Fixes # (issue)

### Checklist

- [x] 填写 PR 描述及相关 issue (write PR description and related issue)
- [x] 代码风格检查通过 (code style check passed)
- [x] PR 中包含单元测试 (include unit test)
- [x] 单元测试通过 (unit test passed)
- [x] 本地开发联调环境验证通过 (local development environment verification passed)

### CODE REVIEW

File: src/dashboard/apigateway/apigateway/apis/web/resource_version/views.py [hunk 1]
```diff
@@ -115,20 +118,45 @@ def get_queryset(self):
...
+            "resource_doc_updated_time": resource_docs_updated_time,
+            "resource_backend": resource_backend,
+            "is_schema_v2": instance.is_schema_v2,
+        }
+
+        # 如果传入了stage,返回对应stage的backend_config以及合并资源插件
+        stage_id = self.request.query_params.get("stage_id")
+
+        if stage_id:
+            backend_config = BackendHandler.get_backend_configs_by_gateway_and_stage(request.gateway.id, stage_id)
```
Comment (alex-smile) [inline src/dashboard/apigateway/apigateway/apis/web/resource_version/views.py:152]: => backend_configs

File: src/dashboard/apigateway/apigateway/apis/web/resource_version/views.py [hunk 2]
```diff
@@ -115,20 +118,45 @@ def get_queryset(self):
...
+            context["resource_backend_config"] = backend_config
+            stage_plugin_bindings = PluginBindingHandler.get_stage_plugin_binding(request.gateway.id, stage_id)
+            context["resource_stage_plugin_binding"] = stage_plugin_bindings
```
Comment (alex-smile) [inline src/dashboard/apigateway/apigateway/apis/web/resource_version/views.py:155]: resource_stage_plugin_binding 这个不是 stage 插件绑定吗，为何变量名中有 resource？ stage_plugin_bindings ?

File: src/dashboard/apigateway/apigateway/biz/backend.py [hunk 3]
```diff
@@ -106,3 +106,12 @@ def deletable(backend: Backend) -> bool:
     @staticmethod
     def get_id_to_instance(gateway_id: int) -> Dict[int, Backend]:
         return {backend.id: backend for backend in Backend.objects.filter(gateway_id=gateway_id)}
+
+    @staticmethod
+    def get_backend_configs_by_gateway_and_stage(gateway_id, stage_id):
```
Comment (alex-smile) [inline src/dashboard/apigateway/apigateway/biz/backend.py:111]: 添加类型注解

Comment (alex-smile) [reply src/dashboard/apigateway/apigateway/biz/backend.py:111]: get_backend_configs_by_stage 就可以了吧？

File: src/dashboard/apigateway/apigateway/biz/plugin_binding.py [hunk 4]
```diff
@@ -28,3 +29,11 @@ def delete_by_bindings(gateway_id: int, bindings: List[PluginBinding]):
 
         PluginBinding.objects.filter(gateway_id=gateway_id, id__in=plugin_binding_ids).delete()
         PluginConfig.objects.filter(gateway_id=gateway_id, id__in=plugin_config_ids).delete()
+
+    @staticmethod
+    def get_stage_plugin_binding(gateway_id: int, stage_id: int) -> dict:
```
Comment (alex-smile) [inline src/dashboard/apigateway/apigateway/biz/plugin_binding.py:34]: => get_stage_plugin_bindings ?

File: src/dashboard/apigateway/apigateway/apis/web/resource_version/serializers.py [hunk 5]
```diff
@@ -33,17 +34,85 @@ class Meta:
...
 
 
+class ResourceInfoSLZ(serializers.Serializer):
+    name = serializers.CharField(help_text="资源名称")
+    method = serializers.CharField(help_text="前端请求方法")
+    path = serializers.CharField(help_text="前端请求路径")
+    description = serializers.CharField(help_text="资源描述")
+    description_en = serializers.CharField(help_text="资源英文描述")
+    gateway_labels = serializers.ListSerializer(
+        source="api_labels", child=serializers.IntegerField(), help_text="标签列表"
```
[the remark this answers is not in the document]
Comment (Han-Ya-Jun) [reply src/dashboard/apigateway/apigateway/apis/web/resource_version/serializers.py:44]: done

File: src/dashboard/apigateway/apigateway/apis/web/resource_version/serializers.py [hunk 6]
```diff
@@ -33,17 +34,85 @@ class Meta:
...
+            "config": obj["proxy"]["config"]
+        }
+        backend_id = obj["proxy"].get("backend_id", None)
+        if backend_id:
+            # 后端服务
+            backend_info = {"id": backend_id, "name": self.context["resource_backend"][backend_id].name}
+
+            # 后端服务配置
+            if "resource_backend_config" in self.context:
+                backend_info["config"] = self.context["resource_backend_config"][backend_id].config
```
[the remark this answers is not in the document]
Comment (Han-Ya-Jun) [reply src/dashboard/apigateway/apigateway/apis/web/resource_version/serializers.py:72]: resource_backend_configs done

File: src/dashboard/apigateway/apigateway/apis/web/resource_version/serializers.py [hunk 7]
```diff
@@ -33,17 +34,85 @@ class Meta:
...
+        resource_binding_plugin_set = set()
+
+        for plugin in obj.get("plugins", []):
+            plugin["binding_type"] = PluginBindingScopeEnum.RESOURCE.value
+            resource_binding_plugin_set.add(plugin["type"])
+            plugins.append(plugin)
+
+        for plugin_type, plugin_binding in self.context.get("resource_stage_plugin_binding", {}).items():
+            # 资源绑定同类型插件覆盖环境插件配置
+            if plugin_type in resource_binding_plugin_set:
```
[the remark this answers is not in the document]
Comment (Han-Ya-Jun) [reply src/dashboard/apigateway/apigateway/apis/web/resource_version/serializers.py:95]: 确实，done

File: src/dashboard/apigateway/apigateway/apis/web/resource_version/serializers.py [hunk 8]
```diff
@@ -33,17 +34,85 @@ class Meta:
...
+
+    def get_proxy(self, obj):
+        proxy = {
+            # 后端配置
+            "config": obj["proxy"]["config"]
+        }
+        backend_id = obj["proxy"].get("backend_id", None)
+        if backend_id:
+            # 后端服务
+            backend_info = {"id": backend_id, "name": self.context["resource_backend"][backend_id].name}
```
[the remark this answers is not in the document]
Comment (Han-Ya-Jun) [reply src/dashboard/apigateway/apigateway/apis/web/resource_version/serializers.py:68]: done

File: src/dashboard/apigateway/apigateway/apis/web/resource_version/serializers.py [hunk 9]
```diff
@@ -33,17 +34,85 @@ class Meta:
...
+    def get_plugins(self, obj):
+
+        plugins = []
+
+        # v2 才有plugin数据
+        if not self.context["is_schema_v2"]:
+            return plugins
+
+        resource_binding_plugin_set = set()
+
```
[the remark this answers is not in the document]
Comment (Han-Ya-Jun) [reply src/dashboard/apigateway/apigateway/apis/web/resource_version/serializers.py:87]: done

File: src/dashboard/apigateway/apigateway/apis/web/resource_version/views.py [hunk 10]
```diff
@@ -115,20 +118,45 @@ def get_queryset(self):
...
-
-        # 补充资源文档的更新时间
+        # 查询资源文档的更新时间
         resource_docs_updated_time = ResourceDocVersion.objects.get_doc_updated_time(request.gateway.id, instance.id)
-        for item in data["data"]:
-            item["doc_updated_time"] = resource_docs_updated_time.get(item["id"], {})
 
-        return OKJsonResponse(data=data)
+        # 查询网关后端服务
+        resource_backend = BackendHandler.get_id_to_instance(request.gateway.id)
```
[the remark this answers is not in the document]
Comment (Han-Ya-Jun) [reply src/dashboard/apigateway/apigateway/apis/web/resource_version/views.py:140]: done

File: src/dashboard/apigateway/apigateway/apis/web/resource_version/views.py (hunk 1 above)
Comment (Han-Ya-Jun) [reply src/dashboard/apigateway/apigateway/apis/web/resource_version/views.py:152]: done

File: src/dashboard/apigateway/apigateway/apis/web/resource_version/views.py (hunk 2 above)
Comment (Han-Ya-Jun) [reply src/dashboard/apigateway/apigateway/apis/web/resource_version/views.py:155]: done

File: src/dashboard/apigateway/apigateway/biz/backend.py (hunk 3 above)
Comment (Han-Ya-Jun) [reply src/dashboard/apigateway/apigateway/biz/backend.py:111]: done

File: src/dashboard/apigateway/apigateway/biz/plugin_binding.py (hunk 4 above)
Comment (Han-Ya-Jun) [reply src/dashboard/apigateway/apigateway/biz/plugin_binding.py:34]: done

**Review by wklken — APPROVED** [review]

**Review by zhu327 — APPROVED** [review]

**Pull request merged**